core_lib/src/interface/scribblearea.cpp

```diff
@@ -923,47 +902,30 @@ void ScribbleArea::handleDrawingOnEmptyFrame()
 
         switch (action)
         {
-        case CREATE_NEW_KEY:
-            mEditor->addNewKey();
-            mEditor->scrubTo(frameNumber);  // Refresh timeline.
-
-            // Hack to clear previous frame's content.
-            if (layer->type() == Layer::BITMAP  &&  previousKeyFrame)
-            {
-                auto asBitmapImage = dynamic_cast<BitmapImage *> (previousKeyFrame);
-
-                if (asBitmapImage)
-                {
-                    drawCanvas(frameNumber, asBitmapImage->bounds());
-                }
-            }
-
-            if (layer->type() == Layer::VECTOR)
-            {
-                auto asVectorImage = dynamic_cast<VectorImage *> (previousKeyFrame);
-
-                if (asVectorImage)
-                {
-                    auto copy(*asVectorImage);
-                    copy.selectAll();
-
-                    drawCanvas(frameNumber, copy.getSelectionRect().toRect());
-                }
+        case KEEP_DRAWING_ON_PREVIOUS_KEY:
+        {
+            if (previousKeyFrame == nullptr) {
+                mEditor->addNewKey();
             }
-
             break;
+        }
         case DUPLICATE_PREVIOUS_KEY:
         {
             if (previousKeyFrame)
             {
                 KeyFrame* dupKey = previousKeyFrame->clone();
                 layer->addKeyFrame(frameNumber, dupKey);
-                mEditor->scrubTo(frameNumber);  // Refresh timeline.
+                mEditor->scrubTo(frameNumber);
```

Comment:
I know this was here before but could we remove this line? Layer::addKeyFrame calls scrubTo already.

Reply:
I think you made the same mistake I did. Layer::addkeyframe does not call scrubto, the method Editor::addkeyframe does though. I removed it initially too but then noticed that timeline didn't update as I thought it would, so I added it again :P

Reply:
Yup that's totally what I did. Nevermind 😋